workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -574,12 +578,43 @@ def _increment_reservoir(self) -> Diagnostics:
             return {}
 
     def _apply_reservoir_update_to_state(self) -> Diagnostics:
-        # TODO: handle tendencies
+        # TODO: handle tendencies. Currently the returned tendencies
+        # are only used for diagnostics and are not used in updating state
         if self._reservoir_predict_stepper is not None:
-            [_, diags, state] = self._reservoir_predict_stepper(
-                self._state.time, self._state
+            [
+                tendencies_from_state_prediction,
+                diags,
+                state_updates,
+            ] = self._reservoir_predict_stepper(self._state.time, self._state)
+            (
+                stepper_diags,
+                net_moistening,
+            ) = self._reservoir_predict_stepper.get_diagnostics(
+                self._state, tendencies_from_state_prediction
+            )
+            diags.update(stepper_diags)
+            if self._reservoir_predict_stepper.diagnostic is True:  # type: ignore
```

Comment:
nit: `if self._reservoir_predict_stepper.diagnostic:`

Reply:
Could rename the underlying attribute to `is_diagnostic` to make it clearer that it's a bool